### PR TITLE
ci: docker release workflow

### PR DESCRIPTION
**Pull request by lumir-mrkva**
- introduced [Dockerfile](/Dockerfile) and [swarm stackfile](/stack.yml) to repo
- added workflow to build images and deploy them to the github

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        id: extract_branch
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ secrets.PKG_USER }}
```
Comment (jak-pan) [inline .github/workflows/docker.yml:73]: Should we have this here? Shouldn't we do this in some safer env (like our AMI only after we authenticate this)

Comment (lumir-mrkva) [reply .github/workflows/docker.yml:73]: secret is not accessible to the forks and this workflow is meant to run on push to the master / stable branch

**Comment by jak-pan** [comment]
Is this ready to go in?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	mkdir /data && \
+	chown -R hydra:hydra /data && \
+	ln -s /data /hydra/.local/share/hydra-dx && \
+	rm -rf /usr/bin /usr/sbin
+
+USER hydra
+EXPOSE 30333 9933 9944
+VOLUME ["/data"]
+
+CMD ["/usr/local/bin/hydra-dx","--chain","lerna"]
```
[the remark this answers is not in the document]
Comment (lumir-mrkva) [reply Dockerfile:26]: problem with endpoint is that you cannot override the base (hydra-dx) to run curl for example which is necessary to rotate keys

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,26 @@
+FROM rust:latest as builder
```
[the remark this answers is not in the document]
Comment (lumir-mrkva) [reply Dockerfile:1]: this base image is better maintained and had images for all platforms. its only used for build so extra space taken is not an issue

File: Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (lumir-mrkva) [reply Dockerfile:26]: this way requires little bit more speficifation when you run node, but that should be taken care of in the stackfile

**Review by martinfridrich — APPROVED** [review]

**Pull request merged**